4337/hardhat.config.ts

```diff
@@ -47,7 +47,7 @@ const userConfig: HardhatUserConfig = {
     sources: 'contracts',
   },
   solidity: {
-    compilers: [{ version: primarySolidityVersion, settings: soliditySettings }, { version: '0.6.12' }, { version: '0.5.17' }],
+    compilers: [{ version: primarySolidityVersion, settings: { evmVersion: 'london', ...soliditySettings }}, { version: '0.6.12' }, { version: '0.5.17' }],
```

Comment:
question: out of curiosity, why is specifying `london` EVM version needed?

Reply:
I actually realised that `london` is not the latest one before the Shanghai upgrade, was tricked by some random website. I've adjusted it to Paris here: https://github.com/safe-global/safe-modules/pull/99